### ISSUE TITLE
Behaviour of `toJSON` with `digits=NA` changed from 1.8.4 -> 1.8.5

### ISSUE THREAD

**Issue by andrjohns** [jeroen/jsonlite#420]
The 1.8.5 update appears to have introduced some floating-point-ness issue when converting to JSON with max precision:
```
> packageVersion("jsonlite")
[1] ‘1.8.4’
> jsonlite::toJSON(list(a=c(0.1, 0.2, 0.3)), digits=NA)
{"a":[0.1,0.2,0.3]} 
```

```
> packageVersion("jsonlite")
[1] ‘1.8.5’
> jsonlite::toJSON(list(a=c(0.1, 0.2, 0.3)), digits=NA)
{"a":[0.10000000000000001,0.20000000000000001,0.29999999999999999]} 
```

**Comment by jeroen** [comment]
Fyi this was fixed in jsonlite 1.8.5

**Comment by zeehio** [comment]
> Fyi this was fixed in jsonlite 1.8.5

A minor typo 😉:

According to https://github.com/jeroen/jsonlite/commit/120cca8df83f6f6e6ce26610b113c15e881a95de, this regression was fixed in 1.8.7.

Version 1.8.5 introduced the regression
